[launch] Using TimerAction results in exception

**Required Info:**

- Operating System:
  - Ubuntu 16.04
- Installation type:
  - source
- Version or commit hash:
  - master/HEAD as of 3 Oct 2018 8597466f
- DDS implementation:
  - Fast-RTPS
- Client library (if applicable):
  - 

#### Steps to reproduce issue
`ros2 launch` the launch file below
```python
import os
from launch import LaunchDescription
import launch.actions
import launch_ros.actions

def generate_launch_description():
    return LaunchDescription([
        launch.actions.TimerAction(
            actions = [
                launch_ros.actions.Node( package='demo_nodes_cpp', node_executable='listener', output='screen')
                ], period = 5.0),
    ])
```

#### Expected behavior
The demo_nodes listener node gets launched
#### Actual behavior
`[ERROR] [launch.LaunchService]: Caught exception in launch (see debug for traceback): 'tuple' object has no attribute 'describe_sub_entities'`

#### Additional information
The full backtrace is here
```
[DEBUG] [launch]: emitting event synchronously: 'launch.events.IncludeLaunchDescription'
[DEBUG] [launch]: emitting event synchronously: 'launch.events.IncludeLaunchDescription'
[DEBUG] [launch.LaunchService]: processing event: '<launch.events.include_launch_description.IncludeLaunchDescription object at 0x7f239e069ba8>'
[DEBUG] [launch.LaunchService]: processing event: '<launch.events.include_launch_description.IncludeLaunchDescription object at 0x7f239e069ba8>' ✓ '<launch.event_handlers.on_include_launch_description.OnIncludeLaunchDescription object at 0x7f239e069908>'
[DEBUG] [launch.LaunchService]: processing event: '<launch.events.include_launch_description.IncludeLaunchDescription object at 0x7f239e069cc0>'
[DEBUG] [launch.LaunchService]: processing event: '<launch.events.include_launch_description.IncludeLaunchDescription object at 0x7f239e069cc0>' ✓ '<launch.event_handlers.on_include_launch_description.OnIncludeLaunchDescription object at 0x7f239e069908>'
[DEBUG] [launch.LaunchService]: Traceback (most recent call last):
  File "/home/crdelsey/src/ros_dev_workspace/navstack_dependencies_ws/build/launch/launch/launch_service.py", line 340, in run
    self.__loop_from_run_thread.run_until_complete(run_loop_task)
  File "/usr/lib/python3.5/asyncio/base_events.py", line 387, in run_until_complete
    return future.result()
  File "/usr/lib/python3.5/asyncio/futures.py", line 274, in result
    raise self._exception
  File "/usr/lib/python3.5/asyncio/tasks.py", line 241, in _step
    result = coro.throw(exc)
  File "/home/crdelsey/src/ros_dev_workspace/navstack_dependencies_ws/build/launch/launch/launch_service.py", line 236, in __run_loop
    await process_one_event_task
  File "/usr/lib/python3.5/asyncio/futures.py", line 361, in __iter__
    yield self  # This tells Task to wait for completion.
  File "/usr/lib/python3.5/asyncio/tasks.py", line 296, in _wakeup
    future.result()
  File "/usr/lib/python3.5/asyncio/futures.py", line 274, in result
    raise self._exception
  File "/usr/lib/python3.5/asyncio/tasks.py", line 239, in _step
    result = coro.send(None)
  File "/usr/lib/python3.5/asyncio/coroutines.py", line 121, in send
    return self.gen.send(value)
  File "/home/crdelsey/src/ros_dev_workspace/navstack_dependencies_ws/build/launch/launch/launch_service.py", line 179, in _process_one_event
    await self.__process_event(next_event)
  File "/usr/lib/python3.5/asyncio/coroutines.py", line 105, in __next__
    return self.gen.send(None)
  File "/home/crdelsey/src/ros_dev_workspace/navstack_dependencies_ws/build/launch/launch/launch_service.py", line 198, in __process_event
    visit_all_entities_and_collect_futures(entity, self.__context))
  File "/home/crdelsey/src/ros_dev_workspace/navstack_dependencies_ws/build/launch/launch/utilities/visit_all_entities_and_collect_futures_impl.py", line 45, in visit_all_entities_and_collect_futures
    futures_to_return += visit_all_entities_and_collect_futures(sub_entity, context)
  File "/home/crdelsey/src/ros_dev_workspace/navstack_dependencies_ws/build/launch/launch/utilities/visit_all_entities_and_collect_futures_impl.py", line 45, in visit_all_entities_and_collect_futures
    futures_to_return += visit_all_entities_and_collect_futures(sub_entity, context)
  File "/home/crdelsey/src/ros_dev_workspace/navstack_dependencies_ws/build/launch/launch/utilities/visit_all_entities_and_collect_futures_impl.py", line 38, in visit_all_entities_and_collect_futures
    sub_entities = entity.visit(context)
  File "/home/crdelsey/src/ros_dev_workspace/navstack_dependencies_ws/build/launch/launch/actions/include_launch_description.py", line 113, in visit
    declared_launch_arguments = launch_description.get_launch_arguments()
  File "/home/crdelsey/src/ros_dev_workspace/navstack_dependencies_ws/build/launch/launch/launch_description.py", line 105, in get_launch_arguments
    process_entities(self.entities, _conditional_inclusion=conditional_inclusion)
  File "/home/crdelsey/src/ros_dev_workspace/navstack_dependencies_ws/build/launch/launch/launch_description.py", line 103, in process_entities
    entity.describe_conditional_sub_entities(), _conditional_inclusion=True)
  File "/home/crdelsey/src/ros_dev_workspace/navstack_dependencies_ws/build/launch/launch/launch_description.py", line 101, in process_entities
    entity.describe_sub_entities(), _conditional_inclusion=False)
AttributeError: 'tuple' object has no attribute 'describe_sub_entities'

[ERROR] [launch.LaunchService]: Caught exception in launch (see debug for traceback): 'tuple' object has no attribute 'describe_sub_entities'
[DEBUG] [launch]: emitting event: 'launch.events.Shutdown'
[DEBUG] [launch.LaunchService]: processing event: '<launch.events.shutdown.Shutdown object at 0x7f23a495ccc0>'
[DEBUG] [launch.LaunchService]: processing event: '<launch.events.shutdown.Shutdown object at 0x7f23a495ccc0>' ✓ '<launch.event_handlers.on_shutdown.OnShutdown object at 0x7f239e0690b8>'
[DEBUG] [launch.LaunchService]: processing event: '<launch.events.shutdown.Shutdown object at 0x7f23a495ccc0>' ✓ '<launch.event_handlers.on_shutdown.OnShutdown object at 0x7f23a16ca390>'
[WARNING] [asyncio]: Executing %s took %.3f seconds
```


Also, commenting out the `describe_conditional_sub_entities` method of the `TimerAction` class seems to allow the launch file to work as intended.

--- a/launch/launch/launch_description.py
+++ b/launch/launch/launch_description.py
@@ -103,8 +103,10 @@ class LaunchDescription(LaunchDescriptionEntity):
                         continue
                     process_entities(
                         entity.describe_sub_entities(), _conditional_inclusion=False)
-                    process_entities(
-                        entity.describe_conditional_sub_entities(), _conditional_inclusion=True)
+                    for conditional_sub_entity in entity.describe_conditional_sub_entities():
+                        process_entities(
+                            conditional_sub_entity[1],
+                            _conditional_inclusion=True)
 
         process_entities(self.entities, _conditional_inclusion=conditional_inclusion)
 

--- a/launch/test/launch/test_launch_description.py
+++ b/launch/test/launch/test_launch_description.py
@@ -70,6 +70,19 @@ def test_launch_description_get_launch_arguments():
     la = ld.get_launch_arguments()
     assert len(la) == 0
 
+    # From issue #144: get_launch_arguments was broken when an entitity had conditional
+    # sub entities
+    class EntityWithConditional(LaunchDescriptionEntity):
+
+        def describe_conditional_sub_entities(self):
+            return [('String describing condition', [DeclareLaunchArgument('foo')])]
+
+    ld = LaunchDescription([
+        EntityWithConditional()
+    ])
+    la = ld.get_launch_arguments()
+    assert len(la) == 1
+
 
 def test_launch_description_add_things():
     """Test adding things to the LaunchDescription class."""

